docs/template_plugin/tests/functional/op_reference/select.cpp

```diff
@@ -0,0 +1,139 @@
+        : data_type(data_type),
+          broadcast(broadcast),
+          select_input_pshape(select_input_pshape),
+          select_input(CreateBlob(ngraph::element::boolean, select_input)),
+          if_input_pshape(if_input_pshape),
+          if_input(CreateBlob(data_type, if_input)),
+          else_input_pshape(else_input_pshape),
+          else_input(CreateBlob(data_type, else_input)),
+          expected_output(CreateBlob(data_type, expected_output)) {}
+
+    ngraph::element::Type data_type;
+    ngraph::op::AutoBroadcastSpec broadcast;
+    ngraph::PartialShape select_input_pshape;
+    InferenceEngine::Blob::Ptr select_input;
+    ngraph::PartialShape if_input_pshape;
+    InferenceEngine::Blob::Ptr if_input;
+    ngraph::PartialShape else_input_pshape;
+    InferenceEngine::Blob::Ptr else_input;
+    InferenceEngine::Blob::Ptr expected_output;
+};
+
+class ReferenceSelectLayerTest : public testing::TestWithParam<SelectParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.data_type, params.broadcast, params.select_input_pshape, params.if_input_pshape, params.else_input_pshape);
+        inputData = {params.select_input, params.if_input, params.else_input};
+        refOutData = {params.expected_output};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<SelectParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "data_type=" << param.data_type << "_";
+        result << "broadcast=" << param.broadcast.m_type << "_";
+        result << "select_shape=" << param.select_input_pshape << "_";
+        result << "if_shape=" << param.if_input_pshape << "_";
+        result << "else_shape=" << param.else_input_pshape;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const ngraph::element::Type& data_type, const op::AutoBroadcastSpec& broadcast,
+                                                    const PartialShape& select_pshape, const PartialShape& if_pshape, const PartialShape& else_pshape) {
+        auto A = std::make_shared<op::Parameter>(element::boolean, select_pshape);
+        auto B = std::make_shared<op::Parameter>(data_type, if_pshape);
+        auto C = std::make_shared<op::Parameter>(data_type, else_pshape);
+        return std::make_shared<Function>(std::make_shared<op::v1::Select>(A, B, C, broadcast), ParameterVector {A, B, C});
+    }
+};
+
+TEST_P(ReferenceSelectLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+INSTANTIATE_TEST_SUITE_P(smoke_Select_With_Hardcoded_Refs, ReferenceSelectLayerTest,
+                         ::testing::Values(
+                             // fp32, no brodcasting
+                             SelectParams(element::f32,                                         // if/else/output data type
```

Comment:
Consider builder instead o comments (and c-tor) e.g.: https://github.com/openvinotoolkit/openvino/pull/6623/files#diff-332d725406461a0f91052683c0660e5eee5616601c83aa7e91a5da03bb5f64bdR150-R156

Reply:
Clever approach, but I will stick to comments this time :)